src/xunit.performance.api/SafeTerminateHandler.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Microsoft.Xunit.Performance.Api.Native.Windows;
+using System;
+
+namespace Microsoft.Xunit.Performance.Api
+{
+    internal sealed class SafeTerminateHandler<T> : IDisposable
+        where T : class, IDisposable
+    {
+        public SafeTerminateHandler(Func<T> createCallback)
```

Comment:
Can you please document the reason that you have a create callback vs. taking in the object itself?  I'm assuming that your goal is to make it more obvious that the lifetime of the object is owned by the SafeTerminateHandler instead of other code that might create the object and then dispose of it on its own.